### ISSUE TITLE
[MU4 Issue] Can't adjust line or slur length via keyboard

### ISSUE THREAD

**Issue by shoogle** [musescore/MuseScore#9902]
It is not possible to make lines longer or shorter via the keyboard. Affects "spanner" elements with draggable handles:

- hairpins _(also unable to adjust height of "jaws")_
- slurs & ties _(also unable to adjust curvature)_
- cresc./dim.
- ottavas (8va/8vb)
- pedal markings
- trill lines
- etc.

__Expected behaviour__

1. On **entering** (i.e. adding/inserting) one of the above elements, it should become selected and the **rightmost** handle should gain keyboard focus.

![image](https://user-images.githubusercontent.com/11011881/145008722-2d4d9493-105e-4aa2-8506-47534eb0700b.png)

2. On **navigating** to one of the above elements with Alt+Left/Alt+Right, the **leftmost** handle should gain keyboard focus.

![image](https://user-images.githubusercontent.com/11011881/145008826-1b477a69-facf-4e87-bbfa-e7b9d4f3c46c.png)

3. Pressing the Tab key should move focus clockwise to one of the other handles (Shift+Tab to go anticlockwise). But see suggested change below.

**Actual behaviour**

1. Line elements are not selected when they are entered in the score.
2. The drag handles do not gain keyboard focus during navigation with Alt+Left and Alt+Right (you must press Alt+Shift+E to edit the line).
3. The Tab shortcut moves focus away from the score instead of moving to a different handle. This is sometimes desirable, sometimes not. See suggestion below.

**Suggested change to Tab navigation**

MuseScore 3 used the Tab key to navigate between drag handles in a closed loop, which prevented Tab being used to navigate to other UI elements like the Inspector (see bug [MU#299941](https://musescore.org/en/node/299941)). In MuseScore 4, perhaps we should "break the loop"? This means:

- Pressing **Tab** would still move you to the **next handle (right)**, but when you reach the **rightmost handle**, pressing Tab again would navigate to a UI panel on the **right** of the score (or below it, like the status bar).
- Pressing **Shift+Tab** would still move you to the **previous handle (left)**, but when you reach the **leftmost handle**, pressing Shift+Tab again would navigate to a UI panel on the **left** of the score (e.g. the Palettes).

Basically, each handle would be treated as a separate UI element for Tab and Shift+Tab, and when you run out of handles you then you start to navigate the rest of the UI.

**Comment by wizofaus** [comment]
Can you retest with the latest build, that should mostly be working now, but there may be a few details still not exactly right.

**Comment by shoogle** [comment]
@wizofaus, it looks like most of the line keyboard shortcuts are working since I last tried, perhaps fixed by your PR #8908. Now just the Tab shortcut to move to a different handle remains broken, as well as a default handle selection on entering or navigating with Alt+Right and Alt+Left. I have edited the issue description accordingly.

**Comment by wizofaus** [comment]
Ok I've fixed it now so that the moment a line (or anything with grip handles) is the only selected object, then you can adjust the *current* grip handle with the cursor keys.
The problem though is the "TAB" key is being mapped to two different purposes - one is explicitly handled by the notation view, the other is for the shortuct "nav-next-panel" which doesn't even appear in the shortcuts dialog in order to allow unmapping that key, but essentially after that "nav-next-panel" command activates, you can't use tab any more to change grips.
I don't know what the right solution here is, but presumably what's really wanted is that if tab *has* triggered a grip handle change, it shouldn't also activate the shortcut.  I'm still trying to work out the best way to implement this.

**Comment by DmitryArefiev** [comment]
> Ok I've fixed it now so that the moment a line (or anything with grip handles) is the only selected object, then you can adjust the current grip handle with the cursor keys

Hi @wizofaus , I've tried builds from #10185 (Mac/Win), and it seems to be it's not fixed yet.. 
When **entering slur**, it doesn't select the rightmost handle. When **entering crescendo**, it doesn't leave hairpin selected only:




https://user-images.githubusercontent.com/90187801/150579049-ee36ade5-1936-4965-aac5-fd14f2afb9b0.mp4


.

**Comment by DmitryArefiev** [comment]
Tested  on Win10, Mac12, LinuxMint 20.2 - FIXED in #10401

**Comment by DmitryArefiev** [comment]
@shoogle Please check it briefly (if I missed something)

should be in the next master
